components/tidb_query_datatype/src/codec/mysql/vector.rs

```diff
@@ -176,6 +176,13 @@ impl<'a> VectorFloat32Ref<'a> {
     }
 
     fn index(&self, idx: usize) -> f32 {
+        if idx > self.len() {
```

Comment:
I think it should exist. Our previous work was to avoid boundary checking in tasks that require frequent computation, so in the function we use `index_uncheck`. In other cases, such as `new`, `fmt` we use `index`

Reply:
What I mean is that `index()` performed double checks. The two checks are the same.

Reply:
You forget to remove the code from L187 to L193.

Reply:
<img width="620" alt="image" src="https://github.com/user-attachments/assets/1c83b482-6643-4e67-948e-2a6f34d8fcbb">

Reply:
sry, I removed it😢